src/main/java/org/javarosa/form/api/FormController.java

```diff
@@ -261,31 +272,32 @@ public int stepToPreviousEvent() {
     }
 
     /**
-     * Retrieves the index of the Group that is the host of a given field list.
+     * Retrieves the index of the Group which hosts child index and is marked with given appearanceTag, null otherwise
      */
-    private FormIndex getFieldListHost(FormIndex child) {
+    @Nullable
+    private FormIndex getHostWithAppearance(FormIndex child, String appearanceTag) {
         int event = mFormEntryController.getModel().getEvent(child);
 
         if (event == FormEntryController.EVENT_QUESTION || event == FormEntryController.EVENT_GROUP || event == FormEntryController.EVENT_REPEAT) {
             // caption[0..len-1]
             // caption[len-1] == the event itself
             // caption[len-2] == the groups containing this group
-            FormEntryCaption[] captions = mFormEntryController.getModel().getCaptionHierarchy();
+            FormEntryCaption[] captions = mFormEntryController.getModel().getCaptionHierarchy(child);
```

Comment:
What does this change represent?

Reply:
if we don't pass the `child` index it finds the captions for the current index in form. Though we want to get `captions` for the `child` index passed according to function def. It was working earlier as we were always passing the `child` as current index.